### PR TITLE
[Docs][Next.js]Fix step for deploying Next.js items to Sitecore [479849]

### PR DESCRIPTION
**Pull request by ancaemcken**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update (non-breaking change; modified files are limited to the `/docs` directory)

### CODE REVIEW

File: docs/data/routes/docs/nextjs/getting-started-nextjs/walkthrough-jsscreate/en.md [hunk 1]
```diff
@@ -75,5 +75,5 @@ Sitecore requires Windows, but the instance can be on a virtual machine or a rem
 To connect your application to Sitecore, you must:
 
 1. [Setup JSS Server Components](/docs/client-frameworks/getting-started/jss-server-install).
-2. [Deploy the application to a Sitecore environment](/docs/client-frameworks/getting-started/app-deployment).
+2. Run `jss deploy items`.
```
Comment (ambrauer) [inline docs/data/routes/docs/nextjs/getting-started-nextjs/walkthrough-jsscreate/en.md:78]: Much of that linked page is actually necessary. Not sure about the best way to communicate that (I'll leave it up to you :), but here's what is needed for Next.js:

**Configuring the Sitecore server to receive the JSS app**

- [NO] Step 1. Configure your app's site and host name
- [YES] Step 2. API Key
- [YES] Step 3. Verification

**Connecting the JSS app to Sitecore**

- [YES] Step 1: Setup connection information
- [YES] Step 2: Config deployment
- [YES(ISH)] Step 3: App Deployment
		This is where we want to callout that _only_ `jss deploy items` is required
- [NO] Step 4: Verification

**Review by ambrauer — CHANGES_REQUESTED** [review]

File: docs/data/routes/docs/nextjs/getting-started-nextjs/walkthrough-jsscreate/en.md (hunk 1 above)
Comment (ancaemcken) [reply docs/data/routes/docs/nextjs/getting-started-nextjs/walkthrough-jsscreate/en.md:78]: Awesome input @ambrauer. Thank you for the clarity.

**Comment by ancaemcken** [comment]
@ambrauer I've detailed the steps as you described and added hash links for each step. I think this gives us what we need. 

I've had to make some changes to the route handler and the link component that renders markdown links so that the hash (id of the heading I link to) gets pushed to history and used to scroll down to the heading. Let me know if I need to fix anything else. Otherwise, if approved, I will cherry-pick this over the weekend to the release.

File: docs/data/routes/docs/nextjs/getting-started-nextjs/walkthrough-jsscreate/en.md [hunk 2]
```diff
@@ -72,8 +72,14 @@ Deploying your code-first artifacts and content to Sitecore, WYSIWYG editing, an
...
 1. [Setup JSS Server Components](/docs/client-frameworks/getting-started/jss-server-install).
-2. [Deploy the application to a Sitecore environment](/docs/client-frameworks/getting-started/app-deployment).
-3. [Connect to the Experience Editor](/docs/nextjs/experience-editor/walkthrough).
+2. Configure the Sitecore server to receive the JSS app: 
+   1. [Obtain a Sitecore API key](/docs/client-frameworks/getting-started/app-deployment#step-2-api-key).
+   2. [Verify the API key is working](/docs/client-frameworks/getting-started/app-deployment#step-3-verification).
+3. Deploy the application to Sitecore: 
+   1. [Setup connection information](/docs/client-frameworks/getting-started/app-deployment#step-1-setup-connection-information). 
+   2. [Deploy the configuration](/docs/client-frameworks/getting-started/app-deployment#step-2-config-deployment).
+   3. Run `jss deploy items`.
```
Comment (ambrauer) [inline docs/data/routes/docs/nextjs/getting-started-nextjs/walkthrough-jsscreate/en.md:84]: Realizing we need to include the `--includeContent` and `--includeDictionary` flags as well here
```suggestion
   3. Run `jss deploy items --includeContent --includeDictionary`.
```

**Review by ambrauer — APPROVED** [review]
Just the one change I realized we need to make for the `jss deploy items` command (add'l flags). Otherwise, this looks great. Thanks for fixing the hash link issue!

**Pull request merged**